spring-pulsar-test/src/main/java/org/springframework/pulsar/test/support/PulsarTestConsumer.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.test.support;
+
+import static org.awaitility.Awaitility.await;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.Message;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.SubscriptionInitialPosition;
+import org.awaitility.core.ConditionTimeoutException;
+
+import org.springframework.pulsar.core.PulsarConsumerFactory;
+
+/**
+ * Fluent API, to be used in tests, for consuming messages from Pulsar topics until a
+ * certain {@code Condition} has been met.
+ *
+ * @param <T> the type of the message payload
+ * @author Jonas Geiregat
+ */
+public class PulsarTestConsumer<T> implements TopicStep, SchemaStep, ConditionsStep<T> {
```

Comment:
I think calling this `PulsarConsumerTestUtil` is more consistent w/ the existing other Spring messaging test utils naming.  `PulsarTestConsumer` makes me think it is just a special type of PulsarConsumer.